crates/core/src/protobuf.rs

```diff
@@ -166,7 +167,7 @@ fn to_read_err(status: Status) -> ReadError {
 
 fn to_write_err(status: Status) -> WriteError {
     match status.code() {
-        Code::FailedPrecondition => WriteError::FailedPrecondition(status.to_string()),
+        Code::FailedPrecondition => WriteError::FailedPrecondition(SimpleStatus::from(status).to_string()),
```

Comment:
Can we use `SimpleStatus` for the `ReadError` above as well? We probably need to let `SimpleStatus` implement `Error` to make it work. There are also a few other places where it could be beneficial (e.g. `NodeOperationError` or the `StatusError`).

Reply:
> Can we use SimpleStatus for the ReadError above as well? We probably need to let SimpleStatus implement Error to make it work. 

makes sense.

Reply:
makes sense.